### ISSUE TITLE
Unable to rename anything if PATCH HTTP method is blocked by proxy

### ISSUE THREAD

**Issue by a1ex4** [jupyterlab/jupyterlab#8196]
## Description

I'm running Jupyterlab behind a corporate proxy, which apparently blocks any PATCH request.
Therefore renaming any file or folder doesn't work, only an error dialog is shown, without much indication as to what happened :
![image](https://user-images.githubusercontent.com/12754559/78830402-2ba79d00-79e8-11ea-8420-4857ae168672.png)
The only way to rename anything is then to open up a terminal and move things around, which is really not ideal for users.
I'm not sure what could be a fix, maybe handling the HTTP return code to apply a workaroud and rename anyway ?
The response code was :

> HTTP/ 405 Method Not Allowed

This is related to #6736, but no real answer was provided.
Thanks in advance for the help !

**Comment by telamonian** [comment]
On the surface, this seems more like a "poorly configured firewall" problem than it does like a jupyterlab problem.

Do you know why your firewall is configured to block all PATCH requests in the first place?

**Comment by a1ex4** [comment]
It sure sound like it, but there's still room for improvement on Jupyterlab's side, at least for handling this type of errors and help the users.
Unfortunately I'm not sure I'll get an answer for this question... will see.

On another note, I was literally exploring Darcula's code in order to update it for the newest Jupyterlab version, when out of nowhere I get a notification from a familiar name... 😄 

My goal was to at least have an active issue on this topic, will see if it actually gets somewhere. Thanks for your answer!

**Comment by jasongrout** [comment]
There have been discussions in the past about communicating with the Jupyter server over a single websocket channel.

**Comment by blink1073** [comment]
Yeah at my last job only POST and GET were allowed on some internal platforms.

**Comment by blink1073** [comment]
Ah, and DELETE.  This was enforced by apache config.

**Comment by jasongrout** [comment]
Yeah, unfortunately we might have to compromise with reality here that some environments do not allow the full HTTP standard. It reminds me of a trend to send all communication protocols over http for similar reasons.

**Comment by blink1073** [comment]
FWIW, that same environment didn't support websockets, but that was because it was an old version of apache.

**Comment by telamonian** [comment]
Surprising. I did not know this was somewhat common. Can you configure a firewall to block on any arbitrary header detail?

If we did decide to support a limited API, is a POST/GET only API something we'd want to support optionally, or something we'd want to switch over to completely? I think there's an obvious maintenance advantage to having only one API.

**Comment by telamonian** [comment]
Also, is there any easy way to set up "tunneling" (using ssh or otherwise) to temporarily disguise a PATCH as a POST?

**Comment by jasongrout** [comment]
I think in any case, this conversation is more suited for the jupyter server repo, which maintains these endpoints (or will once we switch to it): https://github.com/jupyter/jupyter_server

CC @echarles @Zsailer

**Comment by echarles** [comment]
I have worked in internal corporate env where Websocket (even on SSL) was not allowed. We had to request on exception to white list the server IP address to make it work.

@a1ex4 We have plan for JupyterLab to run on the new [JupyterServer](http://github.com/jupyter/jupyter_server). We will discuss your point at the next weekly meeting. Feel free to open an issue on the jupyter server repository so we can comment on it.
